lib/internal/modules/esm/loader.js

```diff
@@ -584,21 +599,24 @@ class ESMLoader {
         }
       }
 
-      validateObject(ctx, `${hookErrIdentifier} context`);
+      if (ctx) validateObject(ctx, `${hookErrIdentifier} context`);
+    };
+    const validateOutput = (hookErrIdentifier, output) => {
+      if (typeof output !== 'object') { // [2]
```

Comment:
Can we have tests where hooks return `null` instead of an object?
```suggestion
      if (typeof output !== 'object' || output === null) { // [2]
```